### PR TITLE
python rest API guide

### PR DESCRIPTION
**Pull request by raksiv**
This is currently in a single file

### CODE REVIEW

**Review by tjholm — COMMENTED** [review]
Looks awesome! Providing some guidance on caching might be good, but could also be a separate guide or best practice document.

File: docs/guides/serverless-rest-api-example.md [hunk 1]
```diff
@@ -391,3 +395,369 @@ curl --location --request PUT '{url}' \
...
+async def download_profile_image(ctx):
+    
+    id = ctx.req.params['id']
+
+    photo =  photos.file("images/{id}/photo.png")
+    photo_url = await photo.sign_url(mode=FileMode.READ, expiry=3600)
+
+    ctx.res.headers['Location'] = [photo_url]
+    ctx.res.status = 303
+```
```
[the remark this answers is not in the document]
Comment (raksiv) [reply docs/guides/serverless-rest-api-example.md:735]: Sounds good, I'll make the update to both guides.

File: docs/guides/serverless-rest-api-example.md [hunk 2]
```diff
@@ -391,3 +395,369 @@ curl --location --request PUT '{url}' \
...
+@profile_api.get("/profiles/:id/image/upload")
+async def upload_profile_image(ctx):
+    
+    id = ctx.req.params['id']
+
+    photo =  photos.file("images/{id}/photo.png")
+    photo_url = await photo.sign_url(mode=FileMode.WRITE, expiry=3600)
+
+    ctx.res.body = photo_url
+```
```
[the remark this answers is not in the document]
Comment (raksiv) [reply docs/guides/serverless-rest-api-example.md:706]: Lets sync up so I can get more info about this one

File: docs/guides/serverless-rest-api-example.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (raksiv) [reply docs/guides/serverless-rest-api-example.md:735]: Are you thinking something like this - 

    photo_url = await photo.sign_url(mode=FileMode.WRITE, expiry=3600)

    expires = datetime.utcnow() + timedelta(seconds=(3600))
    expires = expires.strftime("%a, %d %b %Y %H:%M:%S GMT")
    ctx.res.headers['Expires'] = expires

[the remark this answers is not in the document]
Comment (raksiv) [reply docs/guides/serverless-rest-api-example.md:735]: And changing the 303 to 307?

[the remark this answers is not in the document]
Comment (tjholm) [reply docs/guides/serverless-rest-api-example.md]: Not sure if changing the redirect code is required, haven't looked at the behaviour of expires for `303`, worth trying without changing it and monitoring browser to see what happens on subsequent requests.

File: docs/guides/serverless-rest-api-example.md [hunk 3]
```diff
@@ -391,3 +394,369 @@ curl --location --request PUT '{url}' \
...
+## Coding our Profile API
+
+Let's start by initializing our profiles api, create a file named 'profiles.py' within functions and add the following code.
+
+```python
+from uuid import uuid4
+
+from nitric.resources import api, collection, bucket
+from nitric.application import Nitric
+from nitric.api.storage import FileMode
```
Comment (jyecusch) [inline docs/guides/serverless-rest-api-example.md:487]: This import isn't needed, we can use the `file.upload_url()` and `file.download_url()` methods instead

File: docs/guides/serverless-rest-api-example.md [hunk 4]
```diff
@@ -391,3 +394,369 @@ curl --location --request PUT '{url}' \
...
+### Get Signed URL to Upload Profile Image
+
+```python
+@profile_api.get("/profiles/:id/image/upload")
+async def upload_profile_image(ctx):
+
+    id = ctx.req.params['id']
+
+    photo =  photos.file("images/{id}/photo.png")
+    photo_url = await photo.sign_url(mode=FileMode.WRITE, expiry=3600)
```
Comment (jyecusch) [inline docs/guides/serverless-rest-api-example.md:702]:
```suggestion
    photo_url = await photo.upload_url(expiry=3600)
```

File: docs/guides/serverless-rest-api-example.md [hunk 5]
```diff
@@ -391,3 +394,369 @@ curl --location --request PUT '{url}' \
...
+### Get Signed URL to Download Profile Image
+
+```python
+@profile_api.get("/profiles/:id/image/view")
+async def download_profile_image(ctx):
+
+    id = ctx.req.params['id']
+
+    photo =  photos.file("images/{id}/photo.png")
+    photo_url = await photo.sign_url(mode=FileMode.READ, expiry=3600)
```
Comment (jyecusch) [inline docs/guides/serverless-rest-api-example.md:716]:
```suggestion
    photo_url = await photo.download_url(expiry=3600)
```

File: docs/guides/serverless-rest-api-example.md [hunk 6]
```diff
@@ -391,3 +394,369 @@ curl --location --request PUT '{url}' \
...
+You can also directly redirect to the photo url.
+
+```python
+@profile_api.get("/profiles/:id/image/view")
+async def download_profile_image(ctx):
+
+    id = ctx.req.params['id']
+
+    photo =  photos.file("images/{id}/photo.png")
+    photo_url = await photo.sign_url(mode=FileMode.READ, expiry=3600)
```
Comment (jyecusch) [inline docs/guides/serverless-rest-api-example.md:730]:
```suggestion
    photo_url = await photo.download_url(expiry=3600)
```

**Review by jyecusch — CHANGES_REQUESTED** [review]

**Review by tjholm — DISMISSED** [review]

**Review by jyecusch — APPROVED** [review]

**Pull request merged**